external/langchaingo/textsplitter/token_splitter.go

```diff
@@ -0,0 +1,82 @@
+package textsplitter
+
+import (
+	"fmt"
+
+	"github.com/pkoukk/tiktoken-go"
+)
+
+const (
+	// nolint:gosec
+	_defaultTokenModelName    = "gpt-3.5-turbo"
+	_defaultTokenEncoding     = "cl100k_base"
+	_defaultTokenChunkSize    = 512
+	_defaultTokenChunkOverlap = 100
+)
+
+// TokenSplitter is a text splitter that will split texts by tokens.
+type TokenSplitter struct {
+	ChunkSize         int
+	ChunkOverlap      int
+	ModelName         string
+	EncodingName      string
+	AllowedSpecial    []string
+	DisallowedSpecial []string
+}
+
+func NewTokenSplitter(opts ...Option) TokenSplitter {
```

Comment:
The value of this struct is in its methods so I'd return a pointer. There's quite a lot of literature about pointers vs values so the discusion is nuanced. I like the rationale in this [post](https://preslav.me/2024/04/23/two-reasons-to-prefer-struct-pointers-in-golang/)

```suggestion
func NewTokenSplitter(opts ...Option) *TokenSplitter {
```